django/chat/responses.py

```diff
@@ -425,19 +426,39 @@ async def add_files_to_library():
             content_type="text/event-stream",
         )
 
-    response = synthesizer.synthesize(query=input, nodes=source_nodes)
+    if chat.options.qa_answer_mode != "per-source":
+        response = synthesizer.synthesize(query=input, nodes=source_nodes)
+        response_generator = response.response_gen
+        sources = response.source_nodes
+        return StreamingHttpResponse(
```

Comment:
You could probably refactor slightly so "sources" is only defined in 1 place, and the same return statement is used. This would DRY it out even more.

Just pass response_generator=None in the per-source case, and response_replacer=None in the normal case.